pkg/provisioner/ironic/ironic.go

```diff
@@ -1724,39 +1724,60 @@ func (p *ironicProvisioner) PowerOn() (result provisioner.Result, err error) {
 			p.log.Info("waiting for power status to change")
 			return operationContinuing(powerRequeueDelay)
 		}
-		result, err = p.changePower(ironicNode, nodes.PowerOn)
-		switch err.(type) {
-		case nil:
-		case HostLockedError:
-		default:
-			return transientError(errors.Wrap(err, "failed to power on host"))
+		if ironicNode.LastError != "" {
```

Comment:
The problem here is that once a LastError shows up we'll always report it as the error message. and never try again. That's why we need a `force` flag to both `PowerOn` and `PowerOff`.

Reply:
Fixed that.